src/ecpool_sup.erl

```diff
@@ -41,13 +43,13 @@ start_link() ->
 %%--------------------------------------------------------------------
 
 %% @doc Start a pool.
--spec(start_pool(atom(), atom(), list(tuple())) -> {ok, pid()} | {error, term()}).
-start_pool(Pool, Mod, Opts) when is_atom(Pool) ->
+-spec(start_pool(pool_name(), atom(), list(tuple())) -> {ok, pid()} | {error, term()}).
+start_pool(Pool, Mod, Opts) ->
```

Comment:
nit: mb `orelse is_binary(Pool)`?

Reply:
it can be `any()`, except for 2-tuple like `{Name, Index}`.